runtime/stdlib/account.go

```diff
@@ -791,6 +800,236 @@ func newPublicAccountContractsValue(
 	)
 }
 
+const inboxStorageDomain = "inbox"
+
+func accountInboxPublishFunction(
+	gauge common.MemoryGauge,
+	handler EventEmitter,
+	address common.Address,
+	providerValue interpreter.AddressValue,
+) *interpreter.HostFunctionValue {
+	return interpreter.NewHostFunctionValue(
+		gauge,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			value, ok := invocation.Arguments[0].(*interpreter.CapabilityValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			nameValue, ok := invocation.Arguments[1].(*interpreter.StringValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			recipientValue := invocation.Arguments[2].(interpreter.AddressValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			inter := invocation.Interpreter
+			getLocationRange := invocation.GetLocationRange
+
+			handler.EmitEvent(
+				inter,
+				AccountInboxPublishEventType,
+				[]interpreter.Value{
+					providerValue,
+					recipientValue,
+					nameValue,
+					interpreter.NewTypeValue(gauge, value.StaticType(inter)),
+				},
+				getLocationRange,
+			)
+
+			value = value.Transfer(
+				inter,
+				getLocationRange,
+				atree.Address(address),
+				true,
+				nil,
+			).(*interpreter.CapabilityValue)
+
+			recipient := recipientValue.Transfer(
+				inter,
+				getLocationRange,
+				atree.Address(address),
+				true,
+				nil,
+			).(interpreter.AddressValue)
+
+			publishedValue := interpreter.NewPublishedValue(inter, recipient, value)
```

Comment:
Yes, like noted above, `Transfer` would have to handle its children, like e.g. `SomeValue.Transfer`

Reply:
just double-checking: was this added? couldn't find it in the newest commits

Reply:
Oh; I was interpreted @turbolent's comment to mean that this would be handled in the case where we generalized `PublishedValue` to be a container for arbitrary `Value` types in the future. I can submit another PR with this fix if that wasn't the case.